src/Quaternion.jl

```diff
@@ -24,19 +34,103 @@ Base.promote_rule(::Type{Quaternion{T}}, ::Type{S}) where {T <: Real, S <: Real}
 Base.promote_rule(::Type{Quaternion{T}}, ::Type{Complex{S}}) where {T <: Real, S <: Real} = Quaternion{promote_type(T, S)}
 Base.promote_rule(::Type{Quaternion{T}}, ::Type{Quaternion{S}}) where {T <: Real, S <: Real} = Quaternion{promote_type(T, S)}
 
+"""
+    quat(r, [i, j, k])
+
+Convert real numbers or arrays to quaternion. `i, j, k` defaults to zero.
+
+# Examples
+```jldoctest
+julia> quat(7)
+Quaternion{Int64}(7, 0, 0, 0, false)
+
+julia> quat([1, 2, 3])  # This output will be changed in the next breaking release for consistency. (#94)
+Quaternion{Int64}(0, 1, 2, 3, false)
+```
+"""
+quat
+
 quat(p, v1, v2, v3) = Quaternion(p, v1, v2, v3)
 quat(p, v1, v2, v3, n) = Quaternion(p, v1, v2, v3, n)
 quat(x) = Quaternion(x)
 quat(s, a) = Quaternion(s, a)
 
+"""
+    real(T::Type{<:Quaternion})
+
+Return the type that represents the real part of a value of type `T`.
+e.g: for `T == Quaternion{R}`, returns `R`.
+Equivalent to `typeof(real(zero(T)))`.
+
+# Examples
+```jldoctest
+julia> real(Quaternion{Int})
+Int64
+```
+"""
 Base.real(::Type{Quaternion{T}}) where {T} = T
+
+"""
+    real(q::Quaternion)
+
+Return the real part of the quaternion `q`.
+
+See also: [`imag_part`](@ref), [`quat`](@ref), [`isreal`](@ref).
```

Comment:
We have no `isreal` docstring, and I don't think we need it.
```suggestion
See also: [`imag_part`](@ref), [`quat`](@ref)
```

Reply:
Agreed. I was just referring to the docstring of `Base.real`.